packages/angular_devkit/core/src/virtual-fs/host/memory.ts

```diff
@@ -180,7 +180,7 @@ export class SimpleMemoryHost implements Host<{}> {
     path = this._toAbsolute(path);
     if (this._isDirectory(path)) {
       for (const [cachePath] of this._cache.entries()) {
-        if (path.startsWith(cachePath + NormalizedSep)) {
+        if (cachePath.startsWith(path)) {
```

Comment:
If you have a file that starts with the path string but isn’t a folder (in your test create a file named `/subfile`) it will be deleted.

Reply:
I don't think a file will have the same name with a directory name under the same parent path, do you think right?

Reply:
oh, if the file with the same prefix with the path, but with different suffix, it will delete, I see

Reply:
done